datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextViewHolderFactory.kt

```diff
@@ -48,9 +49,25 @@ internal abstract class QuestionnaireItemEditTextViewHolderDelegate(
     textInputEditText.doAfterTextChanged { editable: Editable? ->
       questionnaireItemViewItem.singleAnswerOrNull = getValue(editable.toString())
       questionnaireItemViewItem.questionnaireResponseItemChangedCallback()
+      applyValidationResult(
+        QuestionnaireResponseItemValidator.validate(
+          questionnaireItemViewItem.questionnaireItem,
+          questionnaireItemViewItem.questionnaireResponseItem
+        )
+      )
     }
   }
 
+  private fun applyValidationResult(
+    validationResult: QuestionnaireResponseItemValidator.ValidationResult
+  ) {
+    val validationMessage =
+      validationResult.validationMessages.filter { it != null }.joinToString {
```

Comment:
```suggestion
      validationResult.validationMessages.filterNotNull().joinToString {
```

actually why would you end up with `null`s in the list?

Reply:
@jingtang10 I think yeah, now we will not end up with null values here.